main.go

```diff
@@ -275,6 +277,13 @@ func Run(ctx context.Context) int {
 			}
 			return nil
 		}))
+		go func() {
```

Comment:
Are you sure we don't need any orchestration here? Why not wait for this routine to return and terminate gracefully?